pkg/cmd/healthcheck.go

```diff
@@ -2,25 +2,70 @@ package cmd
 
 import (
 	"context"
+	"fmt"
 	"time"
 
 	"github.com/openshift/microshift/pkg/healthcheck"
 	"github.com/spf13/cobra"
+	"k8s.io/klog/v2"
 )
 
 func NewHealthcheckCommand() *cobra.Command {
 	var timeout time.Duration
+	var custom string
+
+	var namespace string
+	var deployments []string
+	var daemonsets []string
+	var statefulsets []string
 
 	cmd := &cobra.Command{
 		Use:   "healthcheck",
 		Short: "Verify health of the MicroShift",
+		Long: `Verify health of the MicroShift or custom workloads.
+
+Running command without any options performs a healthcheck of MicroShift service and its core workloads.
+
+Checking health of a custom workloads can be achieved in two ways:
+1. Use --namespace option together with --deployments, --daemonsets, and --statefulsets
+   For example:
+      $ microshift healthcheck --namespace openshift-storage --deployments lvms-operator --daemonsets vg-manager
+      $ microshift healthcheck --namespace openshift-ovn-kubernetes --daemonsets ovnkube-master,ovnkube-node
+
+2. Provide --custom option with a JSON argument. It can wait for several namespaces in single execution.
+   The schema is: '{ "NAMESPACE": {"deployments: ["NAMES"...], "daemonsets: ["NAMES"...], "statefulsets: ["NAMES"...]}}'
+   For example:
+      $ microshift healthcheck --custom '{"openshift-storage":{"deployments": ["lvms-operator"], "daemonsets": ["vg-manager"]}, "openshift-ovn-kubernetes":{"daemonsets": ["ovnkube-master", "ovnkube-node"]}}'
+`,
```

Comment:
Json option provides a way to run a healthcheck for multiple namespaces in parallel but the other option is more user friendly if you just need to check a single namespace

Reply:
Should we not use this option for our checks to speed up the overall process?

Reply:
Under what circumstances would someone issue a custom healthcheck command?
MicroShift is already aware of what is core and what is optional (it applies the kustomize manifests so it knows everything that should be up and running when starting), so who is better suited to do these checks than MicroShift itself?

Why would a user try to perform a custom check? It must be for something that is not core and/or optional packages for MicroShift, which falls in the application realm. Should the user not have their own healthcheck, instead of intertwining with the one built for core MicroShift?

This command could even become part of the MicroShift readiness towards systemd.

Reply:
So this is just a convenience thing - currently we have functions.sh which customers can use. This --custom and --namespace is just something to give a migration path.
I don't know if anyone is using function.sh, but if they would need to wait for their DS,deploy,STS, then we can easily provide helpers.